### PR TITLE
Add preliminary support for osx-arm64

### PR DESCRIPTION
**Pull request by ejizba**
### Issue describing the changes in this PR

Related to https://github.com/Azure/azure-functions-core-tools/issues/2834

I was able to do some basic sanity testing (func init, func new, func start) on an M1 Mac for both JS and C# with a build created from this branch. I think that's good enough to get the PR going, but will continue to do more testing.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Comment by ejizba** [comment]
Update on language testing - these work:
- Node.js
- Dotnet
- Dotnet Isolated

And these do not:
- Python, tracked here: https://github.com/Azure/azure-functions-python-worker/issues/915
- PowerShell, tracked here: https://github.com/Azure/azure-functions-powershell-worker/issues/752
- Java: I didn't even try this one because it appears M1 support has only been released on Java 17 so far (hard to find docs, but here's [one post](https://www.reddit.com/r/MacOS/comments/ptfflp/oracle_released_native_apple_silicon_java_jdk_17/)) and our Java worker only supports Java 8 & 11

**Comment by Francisco-Gamino** [comment]
Thank you @ejizba for sending this out! Do you happen to know how much space will this change increase? I am not sure if we have any restrictions on the payload size.

/cc @AnatoliB for FYI

**Comment by ejizba** [comment]
@Francisco-Gamino It results in another zip being uploaded to the functionscdn storage account for each core tools release, but that's about the only "space" change I can think of. It has no effect on the host since it's a core-tools-only change. It also shouldn't affect any user's machines because the arm64 build is it's own separate package

**Review by michaelpeng36 — APPROVED** [review]
LGTM

File: build/Settings.cs [hunk 1]
```diff
@@ -67,7 +69,11 @@ private static string config(string @default = null, [CallerMemberName] string k
                     { "win-x86", _winPowershellRuntimes },
                     { "win-x64", _winPowershellRuntimes },
                     { "linux-x64", new [] { "linux", "linux-x64", "unix", "linux-musl-x64" } },
-                    { "osx-x64", new [] { "osx", "osx-x64", "unix" } }
+                    { "osx-x64", new [] { "osx", "osx-x64", "unix" } },
+                    // NOTE: arm support wasn't added until PowerShell 7.2
```
Comment (Francisco-Gamino) [inline build/Settings.cs:73]: Could you please update the comment to mention that PowerShell 7.0 does not have ARM support?

**Review by Francisco-Gamino — APPROVED** [review]
@ejizba -- One minor comment; otherwise, LGTM.

**Pull request merged**